src/protocol.c

```diff
@@ -91,6 +91,77 @@ check_host_origin(struct lws *wsi) {
     return false;
 }
 
+/* Parse the request URI to retrieve client-specified URI parameters.
+The URI should be in the format: /path?command=<command>&arg1=<arg>&argN=<arg>
+*/
+int
+parse_client_command(struct lws *wsi, struct tty_client *client) {
```

Comment:
1. Currently, the command is executed Immediately after the websocket connection is established, which is before authentication and may be insecure for some command, so we need to change the order.
2. What about put the `command` in the URI path, so that we can map command parameters directly to URI parameters without the trick of `arg1 - argN`?
3. I prefer sending the command and arguments along with the `AuthToken` as json inside the websocket stream.